log/logtest/logtest.go

```diff
@@ -0,0 +1,117 @@
+package logtest
+
+import (
+	"flag"
+	"github.com/google/zoekt/log"
+	"github.com/google/zoekt/log/internal/encoders"
+	"github.com/google/zoekt/log/internal/logger"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+
+	"testing"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+// Init can be used to instantiate the log package for running tests, to be called in
+// TestMain for the relevant package. Remember to call (*testing.M).Run() after initializing
+// the logger!
+//
+// testing.M is an unused argument, used to indicate this function should be called in
+// TestMain.
+func Init(_ *testing.M) {
+	// ensure Verbose is set up
+	testing.Init()
+	flag.Parse()
+	// set reasonable defaults
+	if testing.Verbose() {
+		initLogger(zapcore.DebugLevel)
+	} else {
+		initLogger(zapcore.WarnLevel)
+	}
+}
+
+// InitWithLevel does the same thing as Init, but uses the provided log level to configure
+// the log level for this package's tests, which can be helpful for exceptionally noisy
+// tests.
+//
+// If your loggers are parameterized, you can also use logtest.NoOp to silence output for
+// specific tests.
+func InitWithLevel(_ *testing.M, level log.Level) {
+	initLogger(level.Parse())
+}
+
+func initLogger(level zapcore.Level) {
+	// use empty resource since development does not include it
+	logger.Init(log.Resource{}, level, encoders.OutputConsole, true)
+}
+
+type CapturedLog struct {
+	Time    time.Time
+	Scope   string
+	Level   log.Level
+	Message string
+	Fields  map[string]any
+}
+
+type LoggerOptions struct {
+	// Level configures the minimum log level to output.
+	Level log.Level
+	// FailOnErrorLogs indicates that the test should fail if an error log is output.
+	FailOnErrorLogs bool
+}
+
+func testLogger(t testing.TB, options LoggerOptions) *zap.Logger {
```

Comment:
similar to scopedTestLogger: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/lib/log/logtest/logtest.go?L73

We simply return a new testing core to replace the one created by `logger.Init()`. A second core is wrapped later in Captured()